request/request.go

```diff
@@ -196,6 +196,15 @@ func (r *Request) Size() int {
 func (r *Request) SizeAndDo(m *dns.Msg) bool {
 	o := r.Req.IsEdns0()
 	if o == nil {
+		// ensure remove OPT if initial query was not Edns0 - compliance with https://tools.ietf.org/html/rfc6891#page-10
+		for i := len(m.Extra) - 1; i >= 0; i-- {
+			if m.Extra[i].Header().Rrtype == dns.TypeOPT {
+				m.Extra[i] = m.Extra[len(m.Extra)-1]
```

Comment:
This will reorder the extra. I'm not sure if that violates any rules but typically the extra is ordered to match the answers.

Reply:
I am not aware of detail here. I can manage to keep the order.